ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -382,12 +381,34 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
     std::string objPrefix;
     if ((ret = s3->wtApi->config_get_string(
            s3->wtApi, session, config, "prefix", &objPrefixConf)) == 0)
-        objPrefix = objPrefixConf.str;
+        objPrefix = std::string(objPrefixConf.str, objPrefixConf.len);
     else if (ret != WT_NOTFOUND) {
         std::cerr << "Error: customize_file_system: config parsing for object prefix";
         return (1);
     }
+    /*
+     * Configure the AWS Client configuration. The options that can be set are documented below:
+     * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/client-config.html
+     */
+    Aws::S3Crt::ClientConfiguration awsConfig;
+    awsConfig.throughputTargetGbps = throughputTargetGbps;
+    awsConfig.partSize = partSize;
 
+    /*
+     * Get the AWS region to be used. If no region is provided, the region will default to the
+     * default region configured in the applicable AWS credentials. The allowable values for AWS
+     * region are listed here in the AWS documentation:
+     * http://sdk.amazonaws.com/cpp/api/LATEST/namespace_aws_1_1_region.html
+     */
+    WT_CONFIG_ITEM regionConf;
+    std::string region;
+    if ((ret = s3->wtApi->config_get_string(s3->wtApi, session, config, "region", &regionConf)) ==
+      0)
+        awsConfig.region = std::string(regionConf.str, regionConf.len);
+    else if (ret != WT_NOTFOUND) {
```

Comment:
As mentioned above, I think it should be mandatory to have a location.

Reply:
What should `customize_file_system` return if no region is provided? Should it to be 1? Or EINVAL like on line 372?